code/datums/components/rust.dm

```diff
@@ -0,0 +1,83 @@
+/datum/component/rust
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+	var/atom/parent_atom
+	var/mutable_appearance/rust_overlay
+	var/static/list/typecache_of_valid_types = list(
+		/turf/closed/wall,
+		/turf/closed/wall/r_wall,
+		/turf/open/floor/plating,
+	)
+
+/turf/closed/wall/rust/New()
+	var/atom/T = new /turf/closed/wall(src)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
```suggestion
	var/atom/wall_new = new /turf/closed/wall(src)
```